spec/core/connector_job_spec.rb

```diff
@@ -37,4 +69,93 @@
       expect(expected_job_ids).to eq(actual_job_ids)
     end
   end
+
+  # TODO: #done! example should be made as a shared example and tested by #done!, #cancel! and #error!, with only status and error set differently in each example.
```

Comment:
Hi @artem-shelkovnikov , as `#done!`, `#cancel!` and `#error!` call the same `terminate!` method, most of the test cases (i.e. the current `describe '#done!'`) should be grouped as a shared example, with the only additional test is then when `#cancel!` is called, `canceled_at` is also updated.

I tried to work on the shared example, but it doesn't work. basically, I have difficulty passing proc to the shared example, e.g.  `subject.done!(ingestion_stats, connector_metadata)`. I also tried to put it in `before` block it doesn't work either.

It cost me about one hour, and i decided to add a `TODO` and proceed.